mino/minogrpc/session/addr.go

```diff
@@ -0,0 +1,145 @@
+package session
+
+import (
+	"fmt"
+
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+)
+
+const (
+	orchestratorCode = "O"
+	followerCode     = "F"
+)
+
+// Address is a representation of the network Address of a participant. The
+// overlay implementation requires a difference between an orchestrator and its
+// source address, where the former initiates a protocol and the later
+// participates.
+//
+// See session.wrapAddress for the abstraction provided to a caller external to
+// the overlay module.
+//
+// - implements mino.Address
+type Address struct {
+	orchestrator bool
+	host         string
+}
+
+// NewOrchestratorAddress creates a new address which will be considered as the
+// initiator of a protocol.
+func NewOrchestratorAddress(addr mino.Address) Address {
+	return Address{
+		orchestrator: true,
+		host:         addr.String(),
+	}
+}
+
+// NewAddress creates a new address.
+func NewAddress(host string) Address {
+	return Address{host: host}
+}
+
+// GetDialAddress returns a string formatted to be understood by grpc.Dial()
+// functions.
+func (a Address) GetDialAddress() string {
+	return a.host
```

Comment:
we actually never check that a.host is valid

Reply:
Well if that's the case, it will fail to dial.